src/RPCServer.ts

```diff
@@ -37,6 +37,37 @@ import * as events from './events';
 
 const cleanupReason = Symbol('CleanupReason');
 
+function composeMessage(error: unknown): string {
+  if (
+    typeof error === 'boolean' ||
+    typeof error === 'number' ||
+    typeof error === 'string' ||
+    typeof error === 'bigint' ||
+    typeof error === 'symbol'
+  ) {
+    return `Non-error literal ${String(error)} was thrown`;
+  }
+  if (
+    typeof error === 'object' &&
+    error != null &&
+    'message' in error &&
+    error.message != null
+  ) {
+    return String(error.message);
+  }
+  if (typeof error === 'object' && error?.constructor?.name != null) {
+    return `Non-error object ${error.constructor.name} was thrown`;
+  }
+  // If an object is not serialisable (for example, Object.create(null)), then
+  // an error would be raised when trying to convert it to a string. In that
+  // case, simply avoid serialising the error.
+  try {
+    return `Non-error value ${error} was thrown`;
+  } catch (e) {
+    return 'Non-error value was thrown'
+  }
```

Comment:
Is this the default case? This is a very weird way to check if something can be stringified. It's also really bad to just ignore what the error is. 

It would be better to check the value with `String()` constructor to convert it. And check the `e` error type for the one you expect. The `Object.create(null)` will throw `TypeError` if you try to convert it. We *never* wholesale ignore what an error is.